src/main/java/io/confluent/idesidecar/restapi/models/graph/RealLocalFetcher.java

```diff
@@ -142,11 +150,39 @@ Uni<ConfluentLocalKafkaCluster> withBrokerDetails(ConfluentLocalKafkaCluster clu
     }
   }
 
+  public String resolveSchemaRegistryUri(String connectionId) {
+    var localConfig = connections.getConnectionSpec(connectionId).localConfig();
+    if (localConfig == null || localConfig.schemaRegistryUri() == null) {
+      return CONFLUENT_LOCAL_DEFAULT_SCHEMA_REGISTRY_URI;
+    }
+    return localConfig.schemaRegistryUri(); // may be null if no SR is to be used
```

Comment:
Okay, my suggestion didn't properly handle a blank URI by returning null. I think we need that, right?

```suggestion
    if (localConfig == null || localConfig.schemaRegistryUri() == null) {
      // Use the default SR URI when missing LocalConfig or missing SR URI in that config
      return CONFLUENT_LOCAL_DEFAULT_SCHEMA_REGISTRY_URI;
    }
    var uri = localConfig.schemaRegistryUri();
    if (uri.isBlank()) {
      // Blank URL means no SR
      return null;
    }
    return uri;
```

Reply:
Done.